src/tools/clippy/clippy_lints/src/significant_drop_tightening.rs

```diff
@@ -79,10 +79,11 @@ impl<'tcx> LateLintPass<'tcx> for SignificantDropTightening<'tcx> {
             if apa.counter <= 1 || !apa.has_expensive_expr_after_last_attr {
                 continue;
             }
+            let first_bind_ident = apa.first_bind_ident.unwrap();
```

Comment:
```suggestion
            debug_assert!(apa.first_bind_ident.is_some());
            let Some(first_bind_ident) = apa.first_bind_ident else { return; };
```

Reply:
`debug_assert`s are only for performance-sensitive checks, if something unexpected happens here it's better to have a panic always (unwrap) than only under some build configurations (debug assert).

Reply:
Yes, the intention here is "I think this is impossible (and I'd like to know about it if that's wrong)" which is preferable to "I think this is impossible (but I'll silently bail out early if that's wrong)".